pyiceberg/table/__init__.py

```diff
@@ -3014,3 +3145,35 @@ def _new_field_id(self) -> int:
 
     def _is_duplicate_partition(self, transform: Transform[Any, Any], partition_field: PartitionField) -> bool:
         return partition_field.field_id not in self._deletes and partition_field.transform == transform
+
+
+def ancestors_between(to_snapshot: int, from_snapshot: Optional[int], table: Table) -> Iterable[Snapshot]:
```

Comment:
Should these methods be placed here? Is there a more suitable place for them?

Reply:
This file is getting rather big, how about `snapshots.py`?